### PR TITLE
Add "jsonproto" option to aquery/cquery --output flag.

### LINKED ISSUE #10421: Add JSON output format to aquery
### Description of the problem / feature request:

Add --output=json to `bazel aquery`

### Feature requests: what underlying problem are you trying to solve with this feature?

I recently encountered the need to use aquery to find all the outputs being generated by a particular set of rules. The results needed post-processing to extract the exec_path. We ended up using a bunch of grep/sed. Having a more common text format like JSON would have allowed us to avoid regex-based processing of structured data.

### What operating system are you running Bazel on?

MacOS Mojave 10.14

### What's the output of `bazel info release`?

release 0.26.1

###  Have you found anything relevant by searching the web?

No

### Any other information, logs, or outputs that you want to share?

No

### PR DESCRIPTION
**Pull request by devnev**
Closes #10421

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphProtoOutputFormatterCallback.java [hunk 1]
```diff
@@ -50,6 +52,7 @@ public String formatName() {
   private final OutputType outputType;
   private final ActionGraphDump actionGraphDump;
   private final AqueryActionFilter actionFilters;
+  private final JsonFormat.Printer jsonPrinter = JsonFormat.printer();
```
[the remark this answers is not in the document]
Comment (joeleba) [reply src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphProtoOutputFormatterCallback.java]: Same for the rest of the instances of `jsonPrinter`.

[the remark this answers is not in the document]
Comment (devnev) [reply src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphProtoOutputFormatterCallback.java]: Makes what easier to test?

[the remark this answers is not in the document]
Comment (devnev) [reply src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphProtoOutputFormatterCallback.java]: For the textproto case we call `TextFormat.print()`, a static method that does not permit any injection. Why do we need something else in jsonproto case?

**Comment by devnev** [comment]
@meisterT I added integration tests based off existing ones for textproto output, I'm not sure if what I'm asserting on particularly makes sense though.

File: src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphProtoOutputFormatterCallback.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (joeleba) [reply src/main/java/com/google/devtools/build/lib/query2/aquery/ActionGraphProtoOutputFormatterCallback.java]: Fair enough.

**Comment by devnev** [comment]
Thanks @meisterT, @joeleba. Do I have to do anything else to get the PR merged?

**Comment by joeleba** [comment]
Merged!

**Pull request closed without merging**